setup-local/src/index.js

```diff
@@ -0,0 +1,38 @@
+const core = require('@actions/core');
+const ActionInput = require('./actionInput');
+const BinaryControl = require('./binaryControl');
+const constants = require('../config/constants');
+
+const {
+  ALLOWED_INPUT_VALUES: {
+    LOCAL_TESTING,
+  },
+} = constants;
+
+/**
+ * Entry point to initiate the Action.
+ * 1. Triggers parsing of action input values
+ * 2. Decides requirement of Local Binary
+ * 3. Start/Stop Local Binary if required
+ * 4. Triggers uploading of artifacts
+ */
+const run = async () => {
+  try {
+    const inputParser = new ActionInput();
+    const stateForBinary = inputParser.getInputStateForBinary();
+    const binaryControl = new BinaryControl(stateForBinary);
+
+    if (stateForBinary.localTesting === LOCAL_TESTING.START) {
+      inputParser.setEnvVariables();
+      await binaryControl.downloadBinary();
+      await binaryControl.startBinary();
+    } else {
+      await binaryControl.stopBinary();
```

Comment:
This will trigger the stop operation. It is a fair assumption that the user will star the binary at least once before triggering the stop.
If the user is trying to stop the binary without even starting, that's an undesirable step and we won't be handling that explicitly.
But if the user had started the binary once, we shall try to stop it. We continue the workflow then.

Reply:
resolving this.